File: Algorithm/Programmers/Python/Level 1/최대공약수와 최소공배수.py
def solution(n, m):
    aa = a(n, m)
    bb = int(n * m / aa)
    return [aa, bb]


def a(n, m):
    mm = max(n, m)
    mn = min(n, m)
    div = mm % mn
    if mm / mn == mm:
        return 1

    if div == 0:
        return mn
    else:
        return a(mn, div)

File: Algorithm/Programmers/Java/Level 1/최대공약수와_최소공배수.java
class 최대공약수와_최소공배수 {
    int get_gcd(int x, int y) {
        for (int i = Math.min(x, y); i > 0; i--) {
            if (x % i == 0 && y % i == 0) {
                return i;
            }
        }
        return 1;
    }

    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        int gcd = get_gcd(n, m);
        answer[0] = gcd;
        answer[1] = n * m / gcd;
        return answer;
    }
}